tokio/src/time/sleep.rs

```diff
@@ -248,76 +248,61 @@ cfg_not_trace! {
 }
 
 impl Sleep {
-    cfg_rt! {
-        #[cfg_attr(not(all(tokio_unstable, feature = "tracing")), allow(unused_variables))]
-        #[track_caller]
-        pub(crate) fn new_timeout(
-            deadline: Instant,
-            location: Option<&'static Location<'static>>,
-        ) -> Sleep {
-            use crate::runtime::Handle;
-
-            let handle = Handle::current().inner;
-            let entry = TimerEntry::new(&handle, deadline);
-
-            #[cfg(all(tokio_unstable, feature = "tracing"))]
-            let inner = {
-                let handle = &handle.time();
-                let time_source = handle.time_source();
-                let deadline_tick = time_source.deadline_to_tick(deadline);
-                let duration = deadline_tick.saturating_sub(time_source.now());
-
-                let location = location.expect("should have location if tracing");
-                let resource_span = tracing::trace_span!(
-                    "runtime.resource",
-                    concrete_type = "Sleep",
-                    kind = "timer",
-                    loc.file = location.file(),
-                    loc.line = location.line(),
-                    loc.col = location.column(),
+    #[cfg_attr(not(all(tokio_unstable, feature = "tracing")), allow(unused_variables))]
+    #[track_caller]
+    pub(crate) fn new_timeout(
+        deadline: Instant,
+        location: Option<&'static Location<'static>>,
+    ) -> Sleep {
+        use crate::runtime::scheduler;
+
+        let handle = scheduler::Handle::current();
```

Comment:
Without this change, the conditional compilation test complains that `scheduler::Handle::current()` is not used when only `time` is enabled. I opted to do the work to update this code here instead of silencing the warning temporarily.